homeassistant/components/unifiprotect/config_flow.py

```diff
@@ -44,8 +48,77 @@ class ProtectFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     def __init__(self) -> None:
         """Init the config flow."""
         super().__init__()
-
         self.entry: config_entries.ConfigEntry | None = None
+        self._discovered_device: dict[str, str] = {}
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle discovery via dhcp."""
+        _LOGGER.debug("Starting discovery via: %s", discovery_info)
+        return await self._async_discovery_handoff()
+
+    async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowResult:
+        """Handle a discovered UniFi device."""
+        _LOGGER.debug("Starting discovery via: %s", discovery_info)
+        return await self._async_discovery_handoff()
+
+    async def _async_discovery_handoff(self) -> FlowResult:
+        """Ensure discovery is active."""
+        # Discovery requires an additional check so we use
```

Comment:
This one is a bit different than usual since we need to pass it off to the discovery library since it can efficiently find devices running the protect app on unifi-os.  For most other integrations just finding the device is sufficient, but for unifiprotect, we don't want to present discovery for devices where the unifiprotect server isn't actually running on the unifi-os device.